BuildResidentialHPXML/resources/schedules.rb

```diff
@@ -418,10 +418,7 @@ def create_stochastic_schedules(args:)
           if j >= @mkc_ts_per_day then break end # break as soon as we have filled activities for the day
 
           transition_probs = transition_matrix[(j - 1) * 7...j * 7] # obtain the transition matrix for current timestep
-          transition_probs_matrix = Matrix[*transition_probs]
-          current_state_vec = Matrix.row_vector(state_vector)
-          state_prob = current_state_vec * transition_probs_matrix # Get a new state_probability array
-          state_prob = state_prob.to_a[0]
+          state_prob = transition_probs[active_state]
```

Comment:
This is obviously so much simpler. Rather than create two matrices and perform matrix multiplication, we simply pull the array of interest out of the 2d array. Reduces time by ~1.5s.

Reply:
Good one!  
The existing code was a result of trying to translate paper-maths into code without thinking throughly :)